github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,550 @@
+package github
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceGithubRepositoryRuleset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryRulesetCreate,
+		Read:   resourceGithubRepositoryRulesetRead,
+		Update: resourceGithubRepositoryRulesetUpdate,
+		Delete: resourceGithubRepositoryRulesetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 100),
+				Description:  "The name of the ruleset.",
+			},
+			"target": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Possible values are `branch` and `tag`.",
+			},
+			"repository": {
```

Comment:
I think we should add an `owner` field to make it possible to explicitly set it. The logic for taking the `owner` field if it's set, or reverting back to the `owner` the provider is configured to use looks something like so:
```go
owner := meta.(*Owner).name
if explicitOwner, ok := d.GetOk("owner"); ok {
    owner = explicitOwner.(string)
}

```

Reply:
Adding this change here: https://github.com/o-sama/terraform-provider-github/pull/1